dev_tools/ci/size-labeler.sh

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env bash
+# Copyright 2025 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Updates size labels on pull requests based on the number of lines changed.
+# Usage:
+#   label-pr-size.sh THE_PR_NUMBER
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[[ "${BASH_VERSINFO[0]}" -ge 4 ]] || { echo "ERROR: Bash version 4+ required." >&2; exit 1; }
+
+set -euo pipefail -o errtrace
+shopt -s inherit_errexit
+
+declare -a LABELS=(
+    "Size: XS"
+    "Size: S"
+    "Size: M"
+    "Size: L"
+    "Size: XL"
+)
+
+declare -A LIMITS=(
+    ["${LABELS[0]}"]=10
+    ["${LABELS[1]}"]=50
+    ["${LABELS[2]}"]=200
+    ["${LABELS[3]}"]=800
+    ["${LABELS[4]}"]="$((2 ** 63 - 1))"
+)
+
+declare -a IGNORED=(
+    "*_pb2.py"
+    "*_pb2.pyi"
+    "*_pb2_grpc.py"
+    ".*.lock"
+)
+
+function info() {
+    echo >&2 "INFO: ${*}"
+}
+
+function error() {
+    echo >&2 "ERROR: ${*}"
+}
+
+function jq_stdin() {
+    local infile
+    infile="$(mktemp)"
+    readonly infile
+
+    cat >"${infile}"
+    jq_file "$@" "${infile}"
+}
+
+function jq_file() {
+    # Regardless of the success, store the return code.
+    # Prepend each sttderr with command args and send back to stderr.
+    jq "${@}" 2> >(echo "$(date -Iseconds) stderr from jq ${*}: " 1>&2) &&
```

Comment:
Let us go back to the original which does timestamp for _every_ line
```suggestion
    jq "${@}" 2> >(ts "stderr from jq ${*}: " 1>&2) &&
```

Reply:
We can't. "ts" doesn't exist in the Ubuntu runners. We'd have to apt install it. It's why I switched to the echo statement.

Reply:
> For future maintenance it is best if we keep the difference between cirq and internal version of the size-labeler.sh at minimum. Installing apt package to GHA workflow is not too hard - [stackoverflow.com/a/58002908/24247414](https://stackoverflow.com/a/58002908/24247414); `ts` is part of the `moreutils` apt package.

Of course, but it's more efficient in terms of CPU cycles and time to skip installing moreutils at every invocation just for this.

More importantly, my understanding is that the internal script won't be used after the internal repos move off GitHub in another month. Is that assumption incorrect? If the internal script will no longer be used, then wouldn't it be better not to be constrained unnecessarily by trying to track it?

In any case, we also have to modify the script in other ways: the labels have to be changed (Cirq uses different capitalization) and the list of ignored items is slightly different.

It would be nice to have the freedom to adapt, enhance this, and make it suitable for all Quantumlib repos. For example, in response to the earlier comment about error messages and the handling of arguments, I expanded the script to handle command-line arguments more completely. (This will be in the next commit in a few moments.)

Reply:
> Of course, but it's more efficient in terms of CPU cycles and time to skip installing moreutils at every invocation just for this.

Good points.  On a second look, `ts` is just adding a constant prefix to every stderr line, there is no time in `ls | ts foo`.  We can use `awk` to do the same, I will push that in shortly.

Reply:
Ah, I was also just now looking at https://stackoverflow.com/a/74938444/743730 and it seems that Bash can print timestamps natively – no need for ts or date or such.

Reply:
Here's a first approximation to `ts`:

```bash
function ts() {
    printf "%(%Y-%m-%dT%H:%M:%S)T %s\n" -1 "$*"
}
```

Reply:
Bash 4's time granularity seems to be limited to seconds.